src/librustc/mir/interpret/eval_context.rs

```diff
@@ -281,6 +281,10 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
         &self,
         instance: ty::InstanceDef<'tcx>,
     ) -> EvalResult<'tcx, &'tcx mir::Mir<'tcx>> {
+        // do not continue if typeck errors occurred
+        if self.tcx.typeck_tables_of(instance.def_id()).tainted_by_errors {
```

Comment:
You have to check if the `def_id` is local, if it's not there can be no errors *and* you can't expect `typeck_tables_of` to work.

Reply:
old const eval does this without checking if it's local: https://github.com/rust-lang/rust/blob/master/src/librustc_const_eval/eval.rs#L787-L793

Reply:
miri works in more cases than the old const eval, which requires HIR inlining still.